src/components/ToolsOverlay/ToolsOverlay.js

```diff
@@ -119,7 +120,8 @@ class ToolsOverlay extends React.PureComponent {
     } = this.props;
 
     const isVisible = (isOpen || true) && !isDisabled;
-    if (!isVisible) {
+    // if translation file is not ready, don't render else it looks weird and bad to see untranslated text
+    if (!isVisible || !tReady) {
```

Comment:
Assuming we are fine with using the `tReady` prop, its possible we may run into future cases like this when we create new components.

I think we should use this at the top level somewhere instead?

Reply:
Yeah we can abstract it in the future. I was thinking we can just hide the text. Instead of disabling the component but this works for now.